tests/bayesian/test_baysian_cp_full.py

```diff
@@ -0,0 +1,231 @@
+            l2ActiveCellsSingleColumn.append(active)
+
+            print
+            print("SENSATION", sensation)
+            print("Target object", simpleObjectRepresentation)
+            print("L2 active", active)
+
+            self.assertTrue(len(simpleObjectRepresentation.difference(active)) == 0,
+                            'Not all object cells were activated missing %s' % simpleObjectRepresentation.difference(
+                                    active))
+
+        active = self.exp1.getL2Representations()[0]
+        self.assertTrue(np.array_equal(active, simpleObjectRepresentation),
+                        '%s additional active cells were made, did not converge' % (len(active) - len(simpleObjectRepresentation)))
+
+    def test_two_unique_toys_l2(self):
+        # 4 sensations in a sequence fixed
+        objects = {
+            "simple": [
+                # location, feature for CC0
+                {0: (set([1, 2, 3]), set([1, 2, 3]))},
+                {0: (set([4, 5, 6]), set([4, 5, 6]))},
+                {0: (set([7, 8, 9]), set([7, 8, 9]))},
+                {0: (set([10, 11, 12]), set([10, 11, 12]))},
+            ],
+            "simple2": [
+                # location, feature for CC0
+                {0: (set([13, 14, 15]), set([13, 14, 15]))},
+                {0: (set([16, 17, 18]), set([16, 17, 18]))},
+                {0: (set([19, 20, 21]), set([19, 20, 21]))},
+                {0: (set([22, 23, 24]), set([22, 23, 24]))},
+            ]
+        }
+        # learn the sensations
+        print "train single column: 4-sequence"
+        self.exp1.learnObjects(objects)
+        self.exp1.sendReset()
+
+        simpleObjectRepresentation = self.exp1.objectL2Representations["simple"][0]
+        simpleObjectRepresentation2 = self.exp1.objectL2Representations["simple2"][0]
+
+        for o in ["simple", "simple2"]:
+
+            sensationStepsSingleColumn = objects[o]
+            objectRepresentation = self.exp1.objectL2Representations[o][0]
+
+            print "inference test: single column - object %s" % o
+            print("Object %s representation" % o, objectRepresentation)
+
+            for i in range(len(sensationStepsSingleColumn)):
+                sensation = sensationStepsSingleColumn[i]
+                self.exp1.infer([sensation], objectName=o, reset=False)
+
+                active = self.exp1.getL2Representations()[0]
+                overlap = simpleObjectRepresentation.intersection(active)
+                overlap2 = simpleObjectRepresentation2.intersection(active)
+
+                print
+                print("Overlap %s %s/%s %s" % ("simple", len(overlap), len(simpleObjectRepresentation), overlap))
+                print("Overlap %s %s/%s %s" % ("simple2", len(overlap2), len(simpleObjectRepresentation2), overlap2))
+                print("L2 active", active)
+
+                self.assertTrue(len(objectRepresentation.difference(active)) == 0,
```

Comment:
Does this need to happen with the first sensation already?

Reply:
yes the sequences are so simple it should simply invoke the object after having it learned successfully. (our CP even worked on that, but for multiple objects with similar features etc it fails -> need other forgetting/learning mechanism for CP)